manager/allocator/cnmallocator/networkallocator.go

```diff
@@ -139,16 +139,39 @@ func (na *cnmNetworkAllocator) Allocate(n *api.Network) error {
 	}
 
 	// No swarm-level allocation can be provided by the network driver for
-	// node-local networks. Only thing needed is populating the driver's name
-	// in the driver's state.
+	// node-local networks. Populate the driver's name, options and also
+	// the IPAM config options, all of these are needed to create
+	// the networks on nodes.
 	if nw.isNodeLocal {
 		n.DriverState = &api.Driver{
 			Name: d.name,
 		}
+
+		if n.Spec.DriverConfig != nil {
+			n.DriverState.Options = n.Spec.DriverConfig.Options
+		}
+
 		// In order to support backward compatibility with older daemon
 		// versions which assumes the network attachment to contains
 		// non nil IPAM attribute, passing an empty object
 		n.IPAM = &api.IPAMOptions{Driver: &api.Driver{}}
+
+		if n.Spec.IPAM != nil {
+			if n.Spec.IPAM.Driver != nil {
+				n.IPAM.Driver.Name = n.Spec.IPAM.Driver.Name
+				n.IPAM.Driver.Options = n.Spec.IPAM.Driver.Options
+			}
+
+			ipamSpec := make([]*api.IPAMConfig, 0, len(n.Spec.IPAM.Configs))
+			for _, ic := range n.Spec.IPAM.Configs {
+				ipamSpec = append(ipamSpec, &api.IPAMConfig{
+					Subnet:  ic.Subnet,
```

Comment:
Isnt subnet config local to node ? It doesnt have to be set at global level here for node local networks. I think waht you need is only options

Reply:
Thanks @abhi  for looking into it.

Yes, subnet config should be local to node, which means every node should have the same subnet. E.g. creating a network like this `docker network create --driver=bridge --subnet=192.168.0.0/24 --scope=swarm test` should result in bridge networks with 192.168.0.0/24 on every node. From the point of view of the network allocator it's a global config, which needs to be passed to every node, like for other non-node local networks also.

Subnet and gateway config is not stored in the options AFAIK, therefore leaving away those configs would result in a bridge network on every node without the specified subnet. It won't also be displayed by docker network inspect run on a master node.